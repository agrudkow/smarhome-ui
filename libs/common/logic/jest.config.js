module.exports = {
  name: 'common-logic',
  preset: '../../../jest.config.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  coverageDirectory: '../../../coverage/libs/common/logic',
};
