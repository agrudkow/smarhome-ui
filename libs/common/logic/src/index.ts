export * from './lib/use-accordion';
export * from './lib/use-window-dimensions';
