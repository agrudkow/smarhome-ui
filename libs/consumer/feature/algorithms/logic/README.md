# consumer-feature-algorithms-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-algorithms-logic` to execute the unit tests via [Jest](https://jestjs.io).
