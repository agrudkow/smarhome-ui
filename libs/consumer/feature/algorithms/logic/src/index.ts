export * from './lib/algorithm.interface';
export * from './lib/algorithms-cells-parser';
export * from './lib/algorithms-data-parser';
