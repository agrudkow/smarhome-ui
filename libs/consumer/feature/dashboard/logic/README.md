# consumer-feature-dashboard-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-dashboard-logic` to execute the unit tests via [Jest](https://jestjs.io).
