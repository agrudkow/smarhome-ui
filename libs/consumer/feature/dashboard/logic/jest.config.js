module.exports = {
  name: 'consumer-feature-dashboard-logic',
  preset: '../../../../../jest.config.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  coverageDirectory:
    '../../../../../coverage/libs/consumer/feature/dashboard/logic',
};
