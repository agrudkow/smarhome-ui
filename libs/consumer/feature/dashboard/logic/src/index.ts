export * from './lib/consumer-feature-dashboard-logic';
