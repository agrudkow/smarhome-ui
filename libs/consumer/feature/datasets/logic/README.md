# consumer-feature-datasets-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-datasets-logic` to execute the unit tests via [Jest](https://jestjs.io).
