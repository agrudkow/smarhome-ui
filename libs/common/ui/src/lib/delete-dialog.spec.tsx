import React from 'react';
import { customRender } from '@smarthome/common/logic';
import DeleteDialog from './delete-dialog';
import { fireEvent } from '@testing-library/react';

describe(' DeleteDialog', () => {
  it('should render successfully', () => {
    const { baseElement } = customRender(
      <DeleteDialog
        open={true}
        title=""
        description=""
        onClose={jest.fn()}
        onConfirm={jest.fn()}
      />
    );
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot', () => {
    const { baseElement } = customRender(
      <DeleteDialog
        open={true}
        title=""
        description=""
        onClose={jest.fn()}
        onConfirm={jest.fn()}
      />
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('should set title', () => {
    const title = 'Test title';
    const { getByText } = customRender(
      <DeleteDialog
        open={true}
        title={title}
        description=""
        onClose={jest.fn()}
        onConfirm={jest.fn()}
      />
    );
    expect(getByText(title)).toBeTruthy();
  });

  it('should set description', () => {
    const description = 'Test description';
    const { getByText } = customRender(
      <DeleteDialog
        open={true}
        description={description}
        title=""
        onClose={jest.fn()}
        onConfirm={jest.fn()}
      />
    );
    expect(getByText(description)).toBeTruthy();
  });

  it('should call handler function on confirm', () => {
    const handleConfirm = jest.fn();
    const { getByText } = customRender(
      <DeleteDialog
        open={true}
        title=""
        description=""
        onClose={jest.fn()}
        onConfirm={handleConfirm}
      />
    );
    fireEvent.click(getByText('Confirm'));
    expect(handleConfirm).toHaveBeenCalledTimes(1);
  });

  it('should call handler function on close rating dailog', () => {
    const handleClose = jest.fn();
    const { getByText } = customRender(
      <DeleteDialog
        open={true}
        title=""
        description=""
        onClose={handleClose}
        onConfirm={jest.fn()}
      />
    );
    fireEvent.click(getByText('Close'));
    expect(handleClose).toHaveBeenCalledTimes(1);
  });
});
