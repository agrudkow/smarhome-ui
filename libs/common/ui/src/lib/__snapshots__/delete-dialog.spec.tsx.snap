// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` DeleteDialog should match snapshot 1`] = `
.c1 {
  color: #555;
  border-color: #555;
  -webkit-transition: color 0.3s ease-in-out,border-color 0.3s ease-in-out;
  transition: color 0.3s ease-in-out,border-color 0.3s ease-in-out;
  font-weight: 550;
  width: 100%;
}

.c1:hover {
  color: #1BBDCE;
  border-color: #1BBDCE;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
}

@media (max-width:480px) {
  .c0 {
    padding: 0;
  }
}

<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiDialog-root"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper"
      role="none presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-labelledby="delete-dialog"
        class="MuiPaper-root MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiPaper-elevation24 MuiPaper-rounded"
        role="dialog"
      >
        <div
          class="MuiDialogTitle-root"
          id="'rating-dialogg-title"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6"
          />
        </div>
        <div
          class="MuiDialogContent-root"
        >
          <p
            class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary"
          />
        </div>
        <div
          class="MuiDialogActions-root MuiDialogActions-spacing"
        >
          <div
            class="c0"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined c1 MuiButton-outlinedPrimary"
              style="margin: 5px 0px 0px;"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Confirm
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined c1 MuiButton-outlinedPrimary"
              style="margin: 5px 0px 0px;"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Close
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
